### PR TITLE
test: rotation

### PR DESCRIPTION
**Pull request by tam7t**
Add an integration test for secret rotation

### CODE REVIEW

File: test/e2e/mount_test.go [hunk 1]
```diff
@@ -289,5 +320,97 @@ func TestMountSyncSecret(t *testing.T) {
...
+	secretFileB := filepath.Join(f.tempDir, "secretValue-B")
+	check(ioutil.WriteFile(secretFileB, secretB, 0644))
+	check(execCmd(exec.Command(
+		"gcloud", "secrets", "versions", "add", f.testRotateSecretID,
+		"--data-file", secretFileB,
+		"--project", f.testProjectID,
+	)))
+
+	// Wait for update.
+	time.Sleep(30 * time.Second)
```
Comment (colinman) [inline test/e2e/mount_test.go:394]: Can we add a comment about the frequency of the update?

**Review by colinman — APPROVED** [review]

File: test/e2e/mount_test.go [hunk 2]
```diff
@@ -289,5 +320,97 @@ func TestMountSyncSecret(t *testing.T) {
...
 func TestMountRotateSecret(t *testing.T) {
-	t.Skip("TODO: test rotate")
+	secretA := []byte("secreta")
+	secretB := []byte("secretb")
+
+	// Enable rotation.
+	check(execCmd(exec.Command("enable-rotation.sh", f.kubeconfigFile)))
+
+	// Wait for deployment to finish.
+	time.Sleep(45 * time.Second)
```
[the remark this answers is not in the document]
Comment (tam7t) [reply test/e2e/mount_test.go:330]: we cannot, the deployement is already 'ready'. i spent a while for a quick way to 1) speedup the deployment or 2) watch the deployment and none of it seemed very script-able.

File: test/e2e/mount_test.go (hunk 1 above)
Comment (tam7t) [reply test/e2e/mount_test.go:395]: Updated

**Review by colinman — APPROVED** [review]

**Pull request merged**